src/main.ts

```diff
@@ -216,11 +274,50 @@ export class Purchases {
           "Creating and returning new instance.",
       );
     }
-    validateApiKey(apiKey);
-    validateAppUserId(appUserId);
-    validateProxyUrl(httpConfig.proxyURL);
-    validateAdditionalHeaders(httpConfig.additionalHeaders);
-    Purchases.instance = new Purchases(apiKey, appUserId, httpConfig, flags);
+
+    let finalApiKey: string;
+    let finalAppUserId: string;
+    let finalHttpConfig: HttpConfig;
+    let finalFlags: FlagsConfig;
+
+    // Check if first argument is a configuration object
+    if (typeof configOrApiKey === "object" && configOrApiKey !== null) {
```

Comment:
Hmm I'm wondering if we should convert the old parameters into a `PurchasesConfig` first, so the rest of the configuration process can be done just with a `PurchasesConfig`, and extract that into a `configureInternal` or something like that... Wdyt? Not a blocker though, since we probably also want to do some runtime validation as you're doing here now.